src/ldp/nn/agent/simple_local_agent.py

```diff
@@ -112,8 +122,30 @@ async def get_asv(
 
         # Update state messages with result and return the new state
         next_state.messages = [*next_state.messages, result.value]
+        self._validate_token_count(next_state.messages, next_state.tools)
```

Comment:
Do we want to validate the output or the input? Why throw an error if the output is over the limit?